ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/EncryptedSecret.scala

```diff
@@ -24,4 +28,40 @@ object EncryptedSecret {
       Base16.encode(authTag), cipherParams)
   }
 
+  implicit object EncryptedSecretEncoder extends Encoder[EncryptedSecret] {
+
+    def apply(secret: EncryptedSecret): Json = {
+      Json.obj(
+        "cipherText" -> secret.cipherText.asJson,
+        "salt" -> secret.salt.asJson,
+        "iv" -> secret.iv.asJson,
+        "authTag" -> secret.authTag.asJson,
+        "cipherParams" -> secret.cipherParams.asJson
+      )
+    }
+
+  }
+
+  implicit object EncryptedSecretDecoder extends Decoder[EncryptedSecret] {
+
+    def apply(cursor: HCursor): Decoder.Result[EncryptedSecret] = {
+      for {
+        cipherText <- cursor.downField("cipherText").as[String]
+        salt <- cursor.downField("salt").as[String]
+        iv <- cursor.downField("iv").as[String]
+        authTag <- cursor.downField("authTag").as[String]
+        cipherParams <- cursor.downField("cipherParams").as[EncryptionSettings]
+      } yield EncryptedSecret(cipherText, salt, iv, authTag, cipherParams)
+    }
+
+  }
+
+//  def main(args: Array[String]): Unit = {
+//    val res = decode[EncryptedSecret](
+//      """{ "cipherText": "", "salt": "", "iv": "", "authTag": "", "cipherParams": { "prf":"abc", "c": 10, "dkLen": 100 }}""")
+//    val jsonRaw = res.right.get.asJson.noSpaces
+//    println(res.toString)
+//    println(jsonRaw)
+//  }
```

Comment:
Do you want to leave it in the code?

Reply:
commented out code should be avoided I guess

Reply:
removed